Source/WebCore/style/calc/StyleCalculationTree.cpp

```diff
@@ -213,6 +189,413 @@ TextStream& operator<<(TextStream& ts, const Tree& tree)
     return ts << tree.root;
 }
 
+// MARK: - Depth Computation
+
+static size_t computeDepth(const Child& root)
+{
+    size_t maximumChildDepth = 0;
+    forAllChildren(root, WTF::makeVisitor(
+        [&](const std::optional<Child>& child) {
+            if (child)
+                maximumChildDepth = std::max(computeDepth(*child), maximumChildDepth);
+        },
+        [&](const Child& child) {
+            maximumChildDepth = std::max(computeDepth(child), maximumChildDepth);
+        },
+        [&](const ChildOrNone& childOrNone) {
+            if (childOrNone.holdsAlternative<Child>())
+                maximumChildDepth = std::max(computeDepth(get<Child>(childOrNone)), maximumChildDepth);
+        },
+        [&](const auto&) {
+            maximumChildDepth = std::max<size_t>(1, maximumChildDepth);
+        }
+    ));
+    return maximumChildDepth + 1;
+}
+
+size_t computeDepth(const Tree& tree)
+{
+    return computeDepth(tree.root);
+}
+
+// MARK: - Simplification
+
+static std::optional<Child> simplify(Sum& root)
+{
+    // This is almost identical to the Sum simplification in CSSCalcTree+Simplification.cpp with the following differences:
```

Comment:
The comment could include some example simplifications to make it less abstract.